### PR TITLE
feat(TDI-38246): add new auth type to Google Storage family

### PR DESCRIPTION
**Pull request by Voievidko**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-38246

**What is the new behavior?**
New auth type was added to all components.
For this all components were rewrite to use new official google storage lib.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tGSGet/tGSGet_begin.javajet [hunk 1]
```diff
@@ -276,3 +297,171 @@ imports="
...
+			%>
+		}
+		com.google.cloud.storage.StorageOptions.Builder builder = com.google.cloud.storage.StorageOptions.newBuilder();
+		storage_<%=cid%> = builder.setCredentials(credential_<%=cid%>).build().getService();
+	<%
+	}
+	if(!getList && !useKeysList){
+		%>
+		com.google.api.gax.paging.Page<com.google.cloud.storage.Bucket> buckets_<%=cid%> = storage_<%=cid%>.list();
+		for (com.google.cloud.storage.Bucket bucket_<%=cid%> : buckets_<%=cid%>.iterateAll()) { //start for
```
Comment (ypiel-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tGSGet/tGSGet_begin.javajet:339]: "//start for" uncomplete comment ?

File: main/plugins/org.talend.designer.components.localprovider/components/tGSGet/tGSGet_begin.javajet [hunk 2]
```diff
@@ -276,3 +297,171 @@ imports="
...
+			if (!"".equals(createDir_<%=cid%>)) {
+				java.io.File parentDir_<%=cid%>=new java.io.File(createDir_<%=cid%>);
+				if(!parentDir_<%=cid%>.exists()){
+					parentDir_<%=cid%>.mkdirs();
+				}
+			}
+			if(!obj_<%=cid%>.isDirectory()){
+				java.io.File parentDir_<%=cid%>=new java.io.File(file_<%=cid%>.getParent());
+				if(!parentDir_<%=cid%>.exists()){
+					parentDir_<%=cid%>.mkdirs();
```
Comment (ypiel-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tGSGet/tGSGet_begin.javajet:453]: What if it returns false ?

File: main/plugins/org.talend.designer.components.libs/libs_src/google-storage-utils/src/main/java/org/talend/gs/util/GSObjectUtil.java [hunk 3]
```diff
@@ -44,6 +46,50 @@
...
+            }
+        }
+        return resultMap;
+    }
+
+    public Map<String, File> generateFileMap(File rootFile, String directory){
+        if (rootFile == null){
+            throw new IllegalArgumentException("File can't be null");
+        }
+        if (directory == null){
```
[the remark this answers is not in the document]
Comment (Voievidko) [reply main/plugins/org.talend.designer.components.libs/libs_src/google-storage-utils/src/main/java/org/talend/gs/util/GSObjectUtil.java:67]: Yes. If string which represent directory is empty that means file will be written right in the root of bucket

File: main/plugins/org.talend.designer.components.localprovider/components/tGSBucketDelete/tGSBucketDelete_main.javajet [hunk 4]
```diff
@@ -99,4 +116,78 @@
...
+		%>
+			log.debug("<%=cid%> - Bucket '" + <%=bucketName%> + "' was deleted.");
+		<%
+		}
+		%>
+	} catch (com.google.cloud.storage.StorageException e_<%=cid%>){
+		<%
+		if(isLog4jEnabled){
+		%>
+			log.error("<%=cid%> - Exception in component <%=cid%>:" + e_<%=cid%>.getMessage());
```
[the remark this answers is not in the document]
Comment (Voievidko) [reply main/plugins/org.talend.designer.components.localprovider/components/tGSBucketDelete/tGSBucketDelete_main.javajet:177]: done

File: main/plugins/org.talend.designer.components.localprovider/components/tGSBucketCreate/tGSBucketCreate_main.javajet [hunk 5]
```diff
@@ -112,5 +130,84 @@
...
+			if (("true").equals(dieOnError)) {
+			%>
+				throw(e_<%=cid%>);
+			<%
+			}
+			%>
+			<%
+			if(isLog4jEnabled){
+			%>
+				log.error("<%=cid%> - Exception in component <%=cid%>:" + e_<%=cid%>.getMessage());
```
[the remark this answers is not in the document]
Comment (Voievidko) [reply main/plugins/org.talend.designer.components.localprovider/components/tGSBucketCreate/tGSBucketCreate_main.javajet:160]: done for all cases

File: main/plugins/org.talend.designer.components.localprovider/components/tGSGet/tGSGet_begin.javajet (hunk 1 above)
Comment (Voievidko) [reply main/plugins/org.talend.designer.components.localprovider/components/tGSGet/tGSGet_begin.javajet:339]: remove comment

File: main/plugins/org.talend.designer.components.localprovider/components/tGSGet/tGSGet_begin.javajet (hunk 2 above)
Comment (Voievidko) [reply main/plugins/org.talend.designer.components.localprovider/components/tGSGet/tGSGet_begin.javajet]: Added error

**Pull request merged**